### PR TITLE
Update apiextensions to v3

### PR DESCRIPTION
**Pull request by rossf7**
## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/controller/chart/resource/status/create.go [hunk 1]
```diff
@@ -81,12 +82,14 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
-			Revision:     releaseContent.Revision,
-			Status:       status,
+			Revision: to.IntP(releaseContent.Revision),
+			Status:   status,
 		},
 		Version: releaseContent.Version,
 	}
+	if !releaseContent.LastDeployed.IsZero() {
+		desiredStatus.Release.LastDeployed = &metav1.Time{Time: releaseContent.LastDeployed}
+	}
```
Comment (rossf7) [inline service/controller/chart/resource/status/create.go:92]: LastDeployed became an optional field in the CRD. So we need to handle nils or the operator crashes.

File: service/controller/chart/resource/status/create.go [hunk 2]
```diff
@@ -130,11 +133,13 @@ func (r *Resource) setStatus(ctx context.Context, cr v1alpha1.Chart, status v1al
...
-		Reason:       status.Reason,
-		Status:       status.Release.Status,
-		Version:      status.Version,
+		AppVersion: status.AppVersion,
+		Reason:     status.Reason,
+		Status:     status.Release.Status,
+		Version:    status.Version,
+	}
+	if status.Release.LastDeployed != nil {
+		request.LastDeployed = *status.Release.LastDeployed
```
Comment (rossf7) [inline service/controller/chart/resource/status/create.go:142]: Also for the webhook request.

**Comment by rossf7** [comment]
Tested on `gauss`.

File: service/controller/chart/resource/status/create.go (hunk 1 above)
Comment (rossf7) [reply service/controller/chart/resource/status/create.go:93]: Ran into a problem in giraffe which is the CRD wasn't up to date and these optional fields were still required.

We should re-run `opsctl ensure crds` for the chart CRD before releasing this.

Comment (tomahawk28) [reply service/controller/chart/resource/status/create.go:93]: Isn't that `chartcrd` resource in app-operator doing that? or is that resource only for TCs?

**Review by tomahawk28 — APPROVED** [review]
LGTM

**Pull request merged**